### DISCUSSION TITLE
bug: base chain type creating typescript error

### DISCUSSION THREAD

**Discussion by gigamesh** [wevm/viem#2232] [Question]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.9.32

### Current Behavior

I have viem & typescript installed in sibling workspaces of a monorepo, with viem and typescript both on their latest versions for every package using them. However I'm getting a type error for `base` chain in one of the workspaces. 

When I command+click on base to see the file in node_modules, both lead to the same file, despite one showing the typescript error and one not:

![image](https://github.com/wevm/viem/assets/32403632/aca48555-da6a-4c00-9a4c-6f389f5eefb7)

### Expected Behavior

No typescript errors

**Comment by gigamesh** [comment]
Update...

Just noticed 2.10.2 was released a few hours ago so I tried upgrading to it and removing the file showing the ts error. 

Despite the typescript server no longer complaining about anything, I still get an error when I try to run files that import the viem config (I'm using `ts-node`): 

![image](https://github.com/wevm/viem/assets/32403632/3298d8bd-576c-4306-87bb-0472106c4396)

**Comment by jxom** [comment]
Moving to discussions as this seems to be a workspace issue rather than a Viem issue.

It might be worth providing a minimal reproduction so others can help you. Screenshots aren’t really useful.

**Reply by gigamesh** [reply]
Minimal repro here: 

https://github.com/gigamesh/viem-ts-error-example

**Reply by tmm** [reply]
Looks fine in this [TS Playground](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbwFBzgBQK4CMA2wDGAwngKYB2MhEZAZsAOYA0Kc+UJAhjCccOTM1RtO3TLgK9+guAAsYMMMwC+cGlAgg4AcgBufEFqShIsRHA5k9ORnCwcAziRuPIeDnBVqN2vSRAB6fBkOYDJ7QyQSAA8TeHxqe3gAOQBRABUAdQB5ACUAaTgAXhYACgBtLTtHLRstFwg3Gu0LKy0AXXN7VgSYAEoAOjoyABMSllQSnV6igD44HSLCwrgwdXwSe3t+8h1+1IANNIB9NABVACEAGQBJQiPUzNy86WmAH1fSkunCueRUVBgMnUAHc4CkoOooGN-v8AAbXSwcPDDOAPbL5eZIjAkABccAAJAhVhB1pttpY9ilDicLjc7minkobNEwCR8NxhnjKg4SFo4K9tPVGvzmpZgDgtLDpKheiwlL0vkhIjFoHEeqxgqFrsMuss-rYeXiqiR+sBhtIhcAOHjLRxTeaWC1xXinTh7cpOt0wjAlSzVV7EnBCAAJACC1ySR2uABEihqQmRtfYygz8m0lTAAJ6soOaxMo5ZZ1kQGgGxz2kVFkgluC2isCqs1132pXxb0rbB4Ih53VwMrG5wkVxWmyujoOAM+5WxSesUgUKi0Bhx-X9nn2tp4-VCPNGnnSuAwKAWeyxPFyBQlYmkrY7frnUMAZRSRxyaDupxyl160iZLDKdZmpuiDjPGoQ2kODQjqBR4nmesjyGAXy-tIZTNkBW6gUECYumK1gwceYTwReSE-nKHoTm2gb2FwwD2HQGxwDkbLQMMAA8LCEHm2rSGIXaSAu1B0PQcAAGRmNh4FwCUjalmAnYSD2vRlGQGAgFgJBQB0ShIDMvoqqYVHwL4IACfAyzCFwPDzjAJT4DZi7CWUIbhpGMZtLKfqGeqJl8RINlxpZogKUQNl2Q5QkMM5YYRlG0YeUgQA)
